src/GraphQL/Subscription/ObservableExtensions.cs

```diff
@@ -0,0 +1,252 @@
+namespace GraphQL.Subscription;
+
+// This file only serves to eliminate the dependency on System.Reactive
+
+internal static class ObservableExtensions
+{
+    /// <summary>
+    /// <para>
+    /// Applies an asynchronous transformation on data events from an <see cref="IObservable{T}"/>.
+    /// Maintains the order of the events produced by the <see cref="IObservable{T}"/>
+    /// whether they are data, error or completion notifications.
+    /// </para>
+    /// <para>
+    /// Ensures that after a subscription has been disposed,
+    /// no more events will be raised (data, error or completion), and signals
+    /// pending asynchronous transformations that a cancellation has been requested.
+    /// </para>
+    /// <para>
+    /// Exceptions passed by the source through <see cref="IObserver{T}.OnError(Exception)"/> or
+    /// generated by <paramref name="transformNext"/> are handled by <paramref name="transformError"/>.
+    /// </para>
+    /// </summary>
+    public static IObservable<TOut> SelectCatchAsync<TIn, TOut>(this IObservable<TIn> observable, Func<TIn, CancellationToken, Task<TOut>> transformNext, Func<Exception, CancellationToken, Task<Exception>> transformError)
```

Comment:
Let's change `observable` to `source`. This is a common naming convention across Rx codebase.

Reply:
And may be change Tasks to ValueTasks as well?

Reply:
Done.

Reply:
Hmm.  I actually think it may be possible here to convert to ValueTask.  Would you aslo change `SubscriptionExecutionStrategy.ProcessDataAsync` to a `ValueTask<ExecutionResult>` as well?

Reply:
And there's three or four methods that deal with exceptions - convert those to ValueTasks also?  Keep in mind that the unhandled exception handler delegate returns a task, but still it may be beneficial to use ValueTasks for the rest of the pipeline.

Reply:
Now I'm talking only about this file. We may change, of course, any part of repo switching to `ValueTask` if appropriate.

Reply:
Done.  Just wanted to double and triple-check that I'm only calling await once on each instance of ValueTask, etc.  All looks good.

Reply:
Should be a good perf win for "high-loaded" subscriptions jumping around all that conversions.

Reply:
I agree it should save a number of allocations.